pr_agent/git_providers/bitbucket_provider.py

```diff
@@ -0,0 +1,110 @@
+import logging
+from datetime import datetime
+from typing import Optional, Tuple
+from urllib.parse import urlparse
+
+import requests
+from atlassian.bitbucket import Cloud
+
+from pr_agent.config_loader import settings
+
+from .git_provider import FilePatchInfo
+
+class BitbucketProvider:
+    def __init__(self, pr_url: Optional[str] = None):
+        s = requests.Session()
+        s.headers['Authorization'] = f'Bearer {settings.get("BITBUCKET.BEARER_TOKEN", None)}'
+        self.bitbucket_client = Cloud(session=s)
+
+        self.workspace_slug = None
+        self.repo_slug = None
+        self.repo = None
+        self.pr_num = None
+        self.pr = None
+        self.temp_comments = []
+        if pr_url:
+            self.set_pr(pr_url)
+
+    def set_pr(self, pr_url: str):
+        self.workspace_slug, self.repo_slug, self.pr_num = self._parse_pr_url(pr_url)
+        self.pr = self._get_pr()
```

Comment:
**Suggestion:** The `set_pr` method in the BitbucketProvider class is currently setting the pull request (PR) directly. It would be better to add error handling to ensure that the PR exists and is accessible. If the PR does not exist or is not accessible, an appropriate error message should be logged and the method should return False. If the PR is set successfully, the method should return True.
```suggestion
    def set_pr(self, pr_url: str) -> bool:
        try:
            self.workspace_slug, self.repo_slug, self.pr_num = self._parse_pr_url(pr_url)
            self.pr = self._get_pr()
            return True
        except Exception as e:
            logging.exception(f'Failed to set PR from URL {pr_url}, error: {e}')
            return False
```